__tests__/app.test.js

```diff
@@ -136,13 +136,81 @@ describe('api/articles/:article_id/comments', () => {
+        }
+        return request(app)
+        .post('/api/articles/1/comments')
+        .send(newComment)
+        .expect(201)
+        .then((response) => {
+            expect(response.body.comment.comment_id).toBe(19)
+            expect(response.body.comment.body).toBe("Itaque quisquam est similique et est perspiciatis reprehenderit voluptatem autem. Voluptatem accusantium eius error adipisci quibusdam doloribus.")
+            expect(response.body.comment.author).toBe('rogersop')
+            expect(response.body.comment.article_id).toBe(1)
+            expect(response.body.comment.votes).toBe(0)
+            expect(response.body.comment).toHaveProperty('created_at')
+        })
+    })
+    test('POST:400 responds with an appropriate status and error message when provided with an imcomplete comment object (no username)', () => {
+        const newComment = {
+            body: "Itaque quisquam est similique et est perspiciatis reprehenderit voluptatem autem. Voluptatem accusantium eius error adipisci quibusdam doloribus."
+        }
+        return request(app)
+          .post('/api/articles/1/comments')
+          .send(newComment)
+          .expect(400)
+          .then((response) => {
+            expect(response.body.msg).toBe('Bad request');
+        });
+    });
+    test('POST:400 responds with an appropriate status and error message when provided with an imcomplete comment object (no body)', () => {
+        const newComment = {
+            username: "rogersop",
+        }
+        return request(app)
+          .post('/api/articles/1/comments')
+          .send(newComment)
+          .expect(400)
+          .then((response) => {
+            expect(response.body.msg).toBe('Bad request');
+        });
+    });
+    test('POST:404 responds with an appropriate status and error message when article_id does not exist', () => {
+        const newComment = {
+            body: "Itaque quisquam est similique et est perspiciatis reprehenderit voluptatem autem. Voluptatem accusantium eius error adipisci quibusdam doloribus.",
+            username: "rogersop",
+        }
+        return request(app)
+          .post('/api/articles/999/comments')
+          .send(newComment)
+          .expect(404)
+          .then((response) => {
+            expect(response.body.msg).toBe('article does not exist');
+        });
+    });
+    test('POST:404 responds with an appropriate status and error message when username does not exist', () => {
+        const newComment = {
+            body: "Itaque quisquam est similique et est perspiciatis reprehenderit voluptatem autem. Voluptatem accusantium eius error adipisci quibusdam doloribus.",
+            username: "test123",
+        }
+        return request(app)
+          .post('/api/articles/1/comments')
+          .send(newComment)
+          .expect(404)
+          .then((response) => {
+            expect(response.body.msg).toBe('article does not exist');
+        });
+    });
```

Comment:
A great start to testing here, however, I would like to also see the following tests added
Status 201, ignores unnecessary properties (such as adding an extra property of 'hobby': 'art'

Status 400, invalid ID, e.g. string of "not-an-id"

Please resub once added